ownCloudAppShared/Tools/AppStatistics.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  AppStatistics.swift
+//  ownCloudAppShared
+//
+//  Created by Michael Neuwert on 10.06.20.
+//  Copyright © 2020 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2020, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import Foundation
+import ownCloudSDK
+import StoreKit
+
+extension TimeInterval {
+    public var days: Int {
+        return Int(self / (3600.0 * 24.0))
+    }
+}
+
+public class AppStatistics {
+
+    enum UserDefaultsKeys: String {
+        case launchDateDefaultsKey, updateDateDefaultsKey, lastVersionDefaultsKey, lastPromptedAboutReviewKey
+    }
+
+    private var launchDate: Date
+    private var updateDate: Date
+
+    private var lastReviewPromptDate: Date? {
+        get {
+            return OCAppIdentity.shared.userDefaults?.object(forKey: UserDefaultsKeys.lastPromptedAboutReviewKey.rawValue) as? Date
+        }
+
+        set {
+            OCAppIdentity.shared.userDefaults?.set(newValue, forKey: UserDefaultsKeys.lastPromptedAboutReviewKey.rawValue)
+            OCAppIdentity.shared.userDefaults?.synchronize()
+        }
+    }
+
+    public static let shared = AppStatistics()
+
+    public var timeIntervalSinceFirstLaunch: TimeInterval {
+        return Date().timeIntervalSince(launchDate)
+    }
+
+    public var timeIntervalSinceUpdate: TimeInterval {
+        return Date().timeIntervalSince(updateDate)
+    }
+
+    private var appVersion: String? {
+        guard let version = OCAppIdentity.shared.appVersion, let build = OCAppIdentity.shared.appBuildNumber else {
```

Comment:
SwiftLint is complaining here (not sure I agree with what it wants, though).

That's what it'd like to see instead:
```suggestion
        guard let version = OCAppIdentity.shared.appVersion, let build = OCAppIdentity.shared.appBuildNumber else { return nil }
```

Reply:
@felix-schwarz SwiftLint didn't complain for me at this one.. I changed it, but I find both variants equally readable and matching our coding style.

Reply:
@felix-schwarz fixed

Reply:
@felix-schwarz Debris is mainly due to use of SwiftLint autocorrect.. And, yes, for the other approach I needed to incorporate fix from other branch into task scheduler which will make it's way into 1.4.